packages/aws-rfdk/lib/deadline/test/secrets-management.test.ts

```diff
@@ -0,0 +1,747 @@
+        (
+          deploymentInstance.userData.addS3DownloadCommand
+            .mockReturnValueOnce('deadlineClientInstallerLocalPath')
+            .mockReturnValueOnce('efsMountScriptLocalPath')
+            .mockReturnValueOnce('directRepoConnectionConfigScriptLocalPath')
+            .mockReturnValueOnce(identityRegistrationSettingsScriptLocalPath)
+        );
+        const clientStack = new Stack(app, 'ClientStack');
+        const client1 = new Construct(clientStack, 'client1');
+        const client2 = new Construct(clientStack, 'client2');
+        createTarget();
+        const baseProps = {
+          registrationStatus: SecretsManagementRegistrationStatus.REGISTERED,
+          role: SecretsManagementRole.CLIENT,
+          vpc,
+          vpcSubnets: { subnetGroupName: DEADLINE_CLIENT_SUBNET_NAME },
+        };
+        target.addSubnetIdentityRegistrationSetting({
+          ...baseProps,
+          dependent: client1,
+        });
+
+        // WHEN
+        target.addSubnetIdentityRegistrationSetting({
+          ...baseProps,
+          dependent: client2,
+        });
+
+        // THEN
+        const resolvedCalls = deploymentInstance.userData.addCommands.mock.calls.map(call => {
+          return deploymentInstanceStack.resolve(call);
+        });
+        const expectedCall = [{
+          'Fn::Join': [
+            '',
+            [
+              // Command is run as "ec2-user" which has the database credentials stored
+              `sudo --login -u ec2-user ${identityRegistrationSettingsScriptLocalPath} `,
+              stack.resolve(Fn.join(
+                ' ',
+                [
+                  '--region',
+                  stack.region,
+                  // The Deadline Secrets Management admin credentials secret ARN is passed
+                  '--credentials',
+                  `"${repository.secretsManagementSettings.credentials!.secretArn}"`,
+                  // The Render Queue's ALB subnets are passed as --connection-subnet args
+                  ...(vpc.selectSubnets({ subnetGroupName: RENDER_QUEUE_ALB_SUBNET_NAME })
+                    .subnetIds.map(subnetID => `--connection-subnet "${subnetID}"`)
+                  ),
+                  // The Deadline Client's subnets, desired role, and registration status are passed as --source-subnet args
+                  ...(vpc.selectSubnets({ subnetGroupName: DEADLINE_CLIENT_SUBNET_NAME })
+                    .subnetIds.map(subnetID => {
+                      return `--source-subnet "${subnetID},${baseProps.role},${baseProps.registrationStatus}"`;
+                    })
+                  ),
+                ],
+              )),
+            ],
+          ],
+        }];
+        expect(resolvedCalls).toContainEqual(expectedCall);
+      });
+
+      // eslint-disable-next-line jest/no-commented-out-tests
+      // test('normalizes subnet ID order', () => {
```

Comment:
How come this test is commented out? I can't understand why subnet ID order matters.

Reply:
Initially I'd thought the order needed to be normalized because I'd thought that changes to the CDK code structure that change the call order to `RenderQueue.configureSecretsManagementAutoRegistration(...)` would cause the user data to change and trigger the configuration to happen again unnecessarily.

This is an optimization and not strictly required for the feature to work correctly. It also turned out that ordering subnet IDs is not always possible when they are unresolved CDK tokens. For this reason, I removed that from the implementation.